python/postprocessing/modules/jme/jetSmearer.py

```diff
@@ -144,13 +138,14 @@ def getSmearValsPt(self, jetIn, genJetIn, rho):
               #         so we would need to randomly "unsmear" the jet, which is impossible
               #
               smearFactor = 1.
+            smear_vals[central_or_shift] = smearFactor
             
+          for central_or_shift in [ enum_nominal, enum_shift_up, enum_shift_down ]:
             # check that smeared jet energy remains positive,
             # as the direction of the jet would change ("flip") otherwise - and this is not what we want
-            if (smearFactor*jet.Perp()) < 1.e-2:
-                smearFactor = 1.e-2
-            smear_vals[central_or_shift] = smearFactor
-        
+            if smear_vals[central_or_shift]*jet.E() < 1.e-2:
+               smear_vals[central_or_shift] = 1.e-2 / jet.E()
+
```

Comment:
Please decrease the indent of this block so that "for central_or_shift in [ enum_nominal, enum_shift_up, enum_shift_down ]:" is in line with "if genJet:" and the corresponding "else:". Currently you are only applying this clipping to the jets without a matched genJet. However, it should be applied to all jets.